core/meta/src/TListOfDataMembers.cxx

```diff
@@ -225,6 +217,8 @@ TDictionary *TListOfDataMembers::Get(DeclId_t id)
 {
    if (!id) return 0;
 
+   R__LOCKGUARD(gInterpreterMutex);
```

Comment:
```suggestion
   TObject *result = THashList::FindObject(name);
   if (result)
       return result;
       
   R__LOCKGUARD(gInterpreterMutex);
```
One of the idea is to avoid taking the global lock if the information is already loaded.  See `TDataMember::Find`

Reply:
See https://github.com/root-project/root/issues/6566

Reply:
Indeed, I mis-remembered whether the access to fIds and its content was made thread-safe by itself ... and it is not.